src/renderHookToSnapshotStream.tsx

```diff
@@ -41,11 +41,21 @@ export interface SnapshotStream<Snapshot, Props> extends Assertable {
    * Does not advance the render iterator.
    */
   waitForNextSnapshot(options?: NextRenderOptions): Promise<Snapshot>
-  rerender: (rerenderCallbackProps: Props) => Promise<void>
+  rerender: (rerenderCallbackProps: VoidOptionalArg<Props>) => Promise<void>
   unmount: () => void
 }
 
-export async function renderHookToSnapshotStream<ReturnValue, Props>(
+/**
+ * if `Arg` can be `undefined`, replace it with `void` to make type represent an optional argument in a function argument position
+ */
+type VoidOptionalArg<Arg> = Arg extends any // distribute members of a potential `Props` union
+  ? undefined extends Arg
+    ? // eslint-disable-next-line @typescript-eslint/no-invalid-void-type
+      void
+    : Arg
+  : Arg
+
+export async function renderHookToSnapshotStream<ReturnValue, Props = void>(
```

Comment:
We can actually reach the same result, but more situational and "correct" by defaulting `Props` to `void` instead of the default `unknown`.

This way if you have an actual argument defined for the function, it will stay required, but if there is no argument it will be `void`, making it optional.

We do need a little bit more work to also correctly work if the function argument is optional from the start - the `VoidOptionalArg` takes care of that.